scripts/on-premises/gcp/gce_image_builder.sh

```diff
@@ -0,0 +1,168 @@
+    $GCLOUD_COMMAND --project $PROJECT \
+        compute instances delete $IMAGE_BUILDER_INSTANCE \
+        --quiet \
+        --zone=$ZONE
+}
+
+exec_command_on_image_builder_instance() {
+    $GCLOUD_COMMAND --project $PROJECT \
+        compute ssh $IMAGE_BUILDER_INSTANCE \
+        --zone=$ZONE \
+        --command="$1"
+}
+
+prepare_image_builder_instance() {
+    case "${DEFAULT_ZONE%%-*}" in
+            "asia")
+                DOWNLOAD_LOCATION=asia
+                ;;
+            "europe")
+                DOWNLOAD_LOCATION=europe
+                ;;
+            *)
+                DOWNLOAD_LOCATION=us
+                ;;
+    esac
+    until exec_command_on_image_builder_instance "echo 'GCE instance is now ready'"
+    do
+        echo 'GCE instance is not ready yet, waiting...'
+        sleep 5
+    done
+    exec_command_on_image_builder_instance "
+        sudo install -m 0755 -d /etc/apt/keyrings &&
+        sudo curl -fsSL https://download.docker.com/linux/debian/gpg \
+            -o /etc/apt/keyrings/docker.asc &&
+        sudo chmod a+r /etc/apt/keyrings/docker.asc &&
+        echo \"deb [arch=$ARCH signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian bookworm stable\" |\
+            sudo tee /etc/apt/sources.list.d/docker.list > /dev/null &&
+        echo \"deb http://deb.debian.org/debian bookworm-backports main\" |\
+            sudo tee /etc/apt/sources.list > /dev/null &&
+        sudo apt update &&
+        sudo apt install -y linux-base -t bookworm-backports &&
+        sudo apt install -y linux-image-cloud-$ARCH -t bookworm-backports &&
+        sudo apt install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin &&
+        sudo docker pull \
+            $DOWNLOAD_LOCATION-docker.pkg.dev/android-cuttlefish-artifacts/cuttlefish-orchestration/cuttlefish-cloud-orchestrator &&
+        sudo docker pull \
+            $DOWNLOAD_LOCATION-docker.pkg.dev/android-cuttlefish-artifacts/cuttlefish-orchestration/cuttlefish-orchestration &&
+        sudo mkdir -p /etc/cloud_orchestrator &&
+        sudo wget -O /etc/cloud_orchestrator/conf.toml \
+            https://artifactregistry.googleapis.com/download/v1/projects/android-cuttlefish-artifacts/locations/$DOWNLOAD_LOCATION/repositories/cloud-orchestrator-config/files/on-premise-single-server:main:conf.toml:download?alt=media &&
+        sudo docker run --restart unless-stopped -d -p 8080:8080 -e CONFIG_FILE="/conf.toml" \
+            -v /etc/cloud_orchestrator/conf.toml:/conf.toml \
+            -v /var/run/docker.sock:/var/run/docker.sock \
+            -t $DOWNLOAD_LOCATION-docker.pkg.dev/android-cuttlefish-artifacts/cuttlefish-orchestration/cuttlefish-cloud-orchestrator:latest
+    "
+}
+
+stop_image_builder_instance() {
+    $GCLOUD_COMMAND --project $PROJECT \
+        compute instances stop $IMAGE_BUILDER_INSTANCE \
+        --zone=$ZONE
+}
+
+create_image() {
+    $GCLOUD_COMMAND --project $PROJECT \
```

Comment:
From my reading it seems you're creating the image out of the boot disk rather than creating the image out of an attached disk.

Creating the image out of the boot disk is problematic as you'll be porting GCE startup information to your image. Alternative is to build the image out of an attached disk. See https://github.com/google/android-cuttlefish/blob/main/tools/baseimage/cmd/create_gce_x86_64_image/main.go for more info,